litestar/utils/helpers.py

```diff
@@ -81,10 +82,9 @@ def unwrap_partial(value: MaybePartial[T]) -> T:
     Returns:
         Callable
     """
-    output: Any = value.func if hasattr(value, "func") else value  # pyright: ignore
-    while hasattr(output, "func"):
-        output = output.func
-    return cast("T", output)
+    if isinstance(value, partial):
```

Comment:
We don't actually need that loop, since `partial` never does the nested wrapping. Internally, it reuses the original func when a `partial` instance is passed.